apolo_kube_client/_core.py

```diff
@@ -113,38 +124,20 @@ async def close(self) -> None:
         logger.info("%s: closed", self)
 
     @property
-    def api_v1_url(self) -> str:
-        return f"{self._base_url}/api/v1"
+    def base_url(self) -> URL:
+        return self._base_url
 
     @property
     def namespace(self) -> str:
         return self._namespace
 
-    @property
-    def namespaces_url(self) -> str:
-        return f"{self.api_v1_url}/namespaces"
-
-    def generate_namespace_url(self, namespace_name: Optional[str] = None) -> str:
-        namespace_name = namespace_name or self._namespace
-        return f"{self.namespaces_url}/{namespace_name}"
-
-    @property
-    def namespace_url(self) -> str:
-        return self.generate_namespace_url()
-
-    def generate_network_policy_url(self, namespace: str) -> str:
-        return (
-            f"{self._base_url}"
-            f"/apis/networking.k8s.io/v1/namespaces"
-            f"/{namespace}/networkpolicies"
-        )
-
     async def request(
         self,
         *args: Any,
         raise_for_status: bool = True,
+        response_type: Any = None,
```

Comment:
Is it possible to avoid changing the return type based on an input argument? I don't think its correct from the typing standpoint. Can there be a method which returns a raw response, while another method which will return a model if needed?